pkg/stats/io_snapshots.go

```diff
@@ -37,25 +42,78 @@ func (h *snapshotHistory) SetInput(id string, latestInput json.Marshaler) {
 	h.latestInput[id] = latestInput
 }
 
-func (h *snapshotHistory) GetInput() ([]byte, error) {
+func (h *snapshotHistory) GetInput(format string, clusters []string, namespaces []string, resourceTypes []string) ([]byte, error) {
 	h.lock.RLock()
 	defer h.lock.RUnlock()
 	if h.latestInput == nil {
 		return []byte{}, nil
 	}
-	result := map[string]interface{}{}
-	for id, obj := range h.latestInput {
-		jsn, err := obj.MarshalJSON()
-		if err != nil {
-			return nil, err
+	genericMaps, err := getGenericMaps(h.latestInput)
+	if err != nil {
+		return nil, err
+	}
+
+	translator := genericMaps["translator"].(map[string]interface{})
+	// Filter resource types
+	if resourceTypes != nil {
+		for k := range translator {
+			if !contains(resourceTypes, k) {
+				delete(translator, k)
+			}
 		}
-		genericMap := map[string]interface{}{}
-		if err := json.Unmarshal(jsn, &genericMap); err != nil {
-			return nil, err
+	}
+
+	// Filter cluster and namespaces
+	if clusters != nil || namespaces != nil {
```

Comment:
Are you sure you want the behavior to be different in the cases where `clusters == nil` and `clusters != nil && len(clusters) == 0`? It seems rare that you would actually want the behavior where you return a result with nothing selected.